dbt_subprojects/dex/models/trades/ronin/platforms/tamadotmeme_v1_ronin_base_trades.sql

```diff
@@ -0,0 +1,119 @@
+    {{wron_token_address}} AS token_sold_address, -- All tokens on tamadot meme are bought using RONIN
+    bet.call_tx_from AS taker,
+    bet.contract_address AS maker,
+    bet.contract_address AS project_contract_address,
+    bet.call_tx_hash AS tx_hash,
+    bet.call_tx_index AS evt_index,
+    bet.call_tx_from AS tx_from,
+    bet.call_tx_to AS tx_to,
+    row_number() over (partition by bet.call_tx_hash, bet.call_tx_index order by bet.call_tx_hash) as duplicates_rank
+
+  FROM  {{ source('tamadotmeme_ronin', 'maincontract_call_buytokenswitheth') }} AS bet
+  LEFT JOIN  {{ source('tamadotmeme_ronin', 'maincontract_evt_tokencreated') }} AS tc
+    ON bet.token = tc.token
+  WHERE call_block_time >= TRY_CAST('{{project_start_date}}' AS TIMESTAMP)
+    {% if is_incremental() %}
+    AND
+        {{ incremental_predicate('bet.call_block_time') }}
+    {% endif %}
+  and call_tx_to!={{edge_case_tx_address}} -- edge case where the tx is both a buy and a sell and coincentally the same token has the same event indiex in respective table
+  and call_success
+),
+
+-- Process "sell" transactions:
+-- Similar to the "buy" section, this CTE normalizes amounts, adds token symbol info,
+-- computes USD value, and applies the same protocol launch filter.
+sell AS (
+  SELECT
+    'ronin' AS blockchain,
+    'tamadotmeme' AS project,
+    '1' AS version,
+    DATE_TRUNC('month', ste.call_block_time) AS block_month,
+    DATE_TRUNC('day', ste.call_block_time) AS block_date,
+    ste.call_block_time AS block_time,
+    ste.call_block_number AS block_number,
+    cast(ste.output_amountOut as double) AS token_bought_amount_raw,
+    cast(ste.amountIn as double) AS token_sold_amount_raw,
+    {{wron_token_address}} AS token_bought_address,  -- All tokens on tamadot meme are sold for RONIN
+    ste.token AS token_sold_address,
+    ste.call_tx_from AS taker,
+    ste.call_tx_to AS maker,
+    ste.contract_address AS project_contract_address,
+    ste.call_tx_hash AS tx_hash,
+    ste.call_tx_index AS evt_index,
+    ste.call_tx_from AS tx_from,
+    ste.call_tx_to AS tx_to,
+    row_number() over(partition by ste.call_tx_hash, ste.call_tx_index order by ste.call_trace_address asc) as duplicates_rank
+
+  FROM  {{ source('tamadotmeme_ronin', 'maincontract_call_selltokensforeth') }} AS ste
+  LEFT JOIN  {{ source('tamadotmeme_ronin', 'maincontract_evt_tokencreated') }} AS tc
+    ON ste.token = tc.token
+  WHERE call_block_time >= TRY_CAST('{{project_start_date}}' AS TIMESTAMP)
+    {% if is_incremental() %}
+    AND
+        {{ incremental_predicate('ste.call_block_time') }}
+    {% endif %}
+  and call_tx_to!={{edge_case_tx_address}} -- edge case where the tx is both a buy and a sell and coincentally the same token has the same event indiex in respective table
+  and call_success
+)
+
+,combined as (
+
+        (SELECT * FROM buy where duplicates_rank=1)
+        UNION ALL
+        (SELECT * FROM sell where duplicates_rank=1)
```

Comment:
what causes these duplicates? is there no other way to filter other than force row number and grab randomly?